### DISCUSSION TITLE
createWriter($showPHPExcel, "Xlsx"); save(file) 2 times crash

### DISCUSSION THREAD

**Discussion by alaindev** [PHPOffice/PhpSpreadsheet#2770] [Q&A]
Hello PhpSpreadsheet Team

I Face a Bug in the library 

IOFactory::load a complexe file.xlsx 
create a IOFactory::createWriter and save it to file2.xlsx 

 IOFactory::load(file2.xlsx) produced by phpspreasheet
create a IOFactory::createWriter and save it to dest.xlsx 

and crash bescause of missing zip stuff ?

Fatal error: Uncaught PhpOffice\PhpSpreadsheet\Writer\Exception: File zip://C:\temp\file2.xlsx# does not exist in C:\Program Files (x86)\EasyPHP-Devserver-17\eds-www\aplitec\includes\modules\phpoffice\phpspreadsheet\src\PhpSpreadsheet\Writer\Xlsx\ContentTypes.php on line 215

any idea ?

Regards
Alain

**Comment by MarkBaker** [comment]
No idea; and with the small amount of information that you've provided here, I can't even try to come up with any ideas.

You provide no information about what might be in your loaded file (other than to say that it is complex). You don't mention what version of PhpSpreadsheet you're running, or what version of PHP. Too little information to do anything other than to agree that there might be a bug somewhere in 250k+ lines of code.

**Comment by alaindev** [comment]
Hello Mark
    Thanks a lot for replying to my call:
    I am using PhpSpreadsheet 1.22
    
    the code is easy as following :
    <?php
        require_once (DIR_WS_MODULES . 'autoload.php'); 
          
          $file = 'c:\tmp\eval-crash.xlsx';
          $showPHPExcel =
      \PhpOffice\PhpSpreadsheet\IOFactory::load($file);
          $sheets_array = $showPHPExcel->getSheetNames();
          $liassewriter =
      \PhpOffice\PhpSpreadsheet\IOFactory::createWriter($showPHPExcel,
      "Xlsx");
          $file = 'c:\tmp\file1.xlsx';
          $liassewriter->save($file);
          
          $step2PHPExcel =
      \PhpOffice\PhpSpreadsheet\IOFactory::load($file);
          $sheets_array = $showPHPExcel->getSheetNames();
          $liassewriter =
      \PhpOffice\PhpSpreadsheet\IOFactory::createWriter($step2PHPExcel,
      "Xlsx");
          $liassewriter->save('c:\tmp\file2.xlsx');
      ?>
    the test 'eval-crash.xlsx' is attached 
    
    the crash seams to appear when image is in footer or not.
    
    Regards
    Alain
    
    
    
    Le 23/04/2022 à 13:22, Mark Baker a
      écrit :
    
    
      
      No idea; and with the small amount of information
        that you've provided here, I can't even tr to come up with any
        ideas.
      You provide no information about what might be in
        your loaded file (other than to say that it is complex). You
        don't mention what version of PhpSpreadsheet you're running, or
        what version of PHP. Too little information to do anything other
        than to agree that there might be a bug somewhere in 250k+ lines
        of code.
      —
        Reply to this email directly, view it on GitHub, or unsubscribe.
        You are receiving this because you authored the thread.Message
          ID: ***@***.***>
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/PHPOffice/PhpSpreadsheet/discussions/2770#discussioncomment-2620389",
"url": "https://github.com/PHPOffice/PhpSpreadsheet/discussions/2770#discussioncomment-2620389",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Reply by alaindev** [reply]
[eval-crash.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/8553132/eval-crash.xlsx)

**Comment by oleibman** [comment]
Solved by PR #3140.
